packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -673,6 +687,83 @@ export const toStyles = (
 				);
 			}
 
+			if ( styleVariationSelectors ) {
+				Object.entries( styleVariationSelectors ).forEach(
+					( [ styleVariationName, styleVariationSelector ] ) => {
+						if ( styles?.variations?.[ styleVariationName ] ) {
+							// If the block uses any custom selectors for block support, add those first.
+							if ( featureSelectors ) {
+								Object.entries( featureSelectors ).forEach(
+									( [ featureName, featureSelector ] ) => {
+										if (
+											styles?.variations?.[
+												styleVariationName
+											]?.[ featureName ]
+										) {
+											const featureStyles = {
+												[ featureName ]:
+													styles.variations[
+														styleVariationName
+													][ featureName ],
+											};
+											const featureDeclarations =
+												getStylesDeclarations(
+													featureStyles
+												);
+											delete styles.variations[
+												styleVariationName
+											][ featureName ];
+
+											if (
+												!! featureDeclarations.length
+											) {
+												const _featureSelectors =
```

Comment:
Tiny nit: Are the underscore prefixes used here to avoid collisions with the `featureSelectors` variable? For readability, would it be worth splitting out any of this logic into a separate function since the nesting here is fairly deep?

Reply:
I'll try and take this bit out to see if it improves readability.